### ISSUE TITLE
[Assignment 4] error occured when integration test

### ISSUE THREAD

**Issue by HLe4s-kaist** [kaist-cp/cs492-uarch#89]
```
java.io.IOException: User limit of inotify instances reached or too many open files
        at java.base/sun.nio.fs.LinuxWatchService.<init>(LinuxWatchService.java:62)
        at java.base/sun.nio.fs.LinuxFileSystem.newWatchService(LinuxFileSystem.java:47)
        at com.swoval.files.RegisterableWatchServices$RegisterableWatchServiceImpl.<init>(RegisterableWatchServices.java:34)
        at com.swoval.files.RegisterableWatchServices.get(RegisterableWatchServices.java:21)
        at com.swoval.files.PlatformWatcher.make(PlatformWatcher.java:11)
        at com.swoval.files.PathWatchers.get(PathWatchers.java:85)
        at com.swoval.files.FileTreeRepositories.get(FileTreeRepositories.java:64)
        at com.swoval.files.FileTreeRepositories.get(FileTreeRepositories.java:32)
        at sbt.internal.nio.FileTreeRepositoryImpl.<init>(FileTreeRepositoryImpl.scala:47)
        at sbt.internal.nio.FileTreeRepository$.default(FileTreeRepository.scala:40)
        at sbt.BuiltinCommands$.$anonfun$setupGlobalFileTreeRepository$1(Main.scala:984)
        at sbt.BuiltinCommands$.$anonfun$doLoadProject$5(Main.scala:976)
        at sbt.Project$.setProject(Project.scala:501)
        at sbt.BuiltinCommands$.doLoadProject(Main.scala:976)
        at sbt.BuiltinCommands$.$anonfun$loadProjectImpl$2(Main.scala:914)
        at sbt.Command$.$anonfun$applyEffect$4(Command.scala:150)
        at sbt.Command$.$anonfun$applyEffect$2(Command.scala:145)
        at sbt.Command$.process(Command.scala:189)
        at sbt.MainLoop$.$anonfun$processCommand$5(MainLoop.scala:245)
        at scala.Option.getOrElse(Option.scala:189)
        at sbt.MainLoop$.process$1(MainLoop.scala:245)
        at sbt.MainLoop$.processCommand(MainLoop.scala:278)
        at sbt.MainLoop$.$anonfun$next$5(MainLoop.scala:163)
        at sbt.State$StateOpsImpl$.runCmd$1(State.scala:289)
        at sbt.State$StateOpsImpl$.process$extension(State.scala:325)
        at sbt.MainLoop$.$anonfun$next$4(MainLoop.scala:163)
        at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
        at sbt.MainLoop$.next(MainLoop.scala:163)
        at sbt.MainLoop$.run(MainLoop.scala:144)
        at sbt.MainLoop$.$anonfun$runWithNewLog$1(MainLoop.scala:119)
        at sbt.io.Using.apply(Using.scala:27)
        at sbt.MainLoop$.runWithNewLog(MainLoop.scala:112)
        at sbt.MainLoop$.runAndClearLast(MainLoop.scala:66)
        at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:51)
        at sbt.MainLoop$.runLogged(MainLoop.scala:42)
        at sbt.StandardMain$.runManaged(Main.scala:223)
        at sbt.xMain$.$anonfun$run$11(Main.scala:133)
        at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
        at scala.Console$.withIn(Console.scala:230)
        at sbt.internal.util.Terminal$.withIn(Terminal.scala:578)
        at sbt.internal.util.Terminal$.$anonfun$withStreams$1(Terminal.scala:358)
        at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
        at scala.Console$.withOut(Console.scala:167)
        at sbt.internal.util.Terminal$.$anonfun$withOut$2(Terminal.scala:568)
        at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
        at scala.Console$.withErr(Console.scala:196)
        at sbt.internal.util.Terminal$.withOut(Terminal.scala:568)
        at sbt.internal.util.Terminal$.withStreams(Terminal.scala:358)
        at sbt.xMain$.withStreams$1(Main.scala:87)
        at sbt.xMain$.run(Main.scala:121)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at sbt.internal.XMainConfiguration.run(XMainConfiguration.java:57)
        at sbt.xMain.run(Main.scala:46)
        at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
        at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
        at xsbt.boot.Launch$.run(Launch.scala:109)
        at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
        at xsbt.boot.Launch$.launch(Launch.scala:117)
        at xsbt.boot.Launch$.apply(Launch.scala:18)
        at xsbt.boot.Boot$.runImpl(Boot.scala:41)
        at xsbt.boot.Boot$.main(Boot.scala:17)
        at xsbt.boot.Boot.main(Boot.scala)
[error] java.io.IOException: User limit of inotify instances reached or too many open files
[error] Use 'last' for the full log.
```
when I type this 
```
python3 scripts/gemmini/main.py build -c pe
```

**Comment by seo-rii** [comment]
Restarting workspace may solve the problem.

**Comment by minseongg** [comment]
Sorry, I cannot reproduce that error for now.

As @seo-rii said, could you please restart the workspace and then try again?

<img width="1439" alt="Restart" src="https://github.com/user-attachments/assets/65c7d4a6-ecaa-4584-bd0c-bf134b47dd2e">

(You may install rustup again after restart the workspace)

**Comment by HLe4s-kaist** [comment]
Thank you all for your answer. 
Actually before I make this issue, I've already been restarting and creating new docker again and again and it did not do a trick. 

But I also found the reason which occured this error. 
I used root user for my docker container, since I didn't want add 'sudo' before my command lines. 
Now I've fully migrated to normal user. After that, this problem doens't occur and I finished assignment 4 😂.
